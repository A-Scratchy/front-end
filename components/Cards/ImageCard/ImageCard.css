.ImageCard {
  flex-flow: row wrap;
  height: 200px;
  justify-content: flex-start;
  padding: 0;
  width: 600px;
}

.image {
  height: 100%;
  width: 50%;
}

.content {
  align-items: center;
  display: flex;
  flex-flow: column;
  overflow: scroll;
  margin: 1rem; /* margin subtracted from width and height */
  height: calc(100% - 2rem); /* 1rem margin top AND bottom hence 2rem */
  width: calc(50% - 2rem); /* 1rem margin left AND right hence 2rem */
}

/* Vertical space between cards for vertical layout */
@media screen and (max-width: 768px) {
  .ImageCard {
    height: 400px;
    width: 300px;
  }

  .image {
    height: 50%;
    width: 100%;
  }

  .content {
    height: 50%;
    width: 100%;
  }
}
