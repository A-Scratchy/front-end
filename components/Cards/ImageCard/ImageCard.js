import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import Button from 'common/components/Button/Button';
import OutboundLink from 'common/components/OutboundLink/OutboundLink';
import styles from './ImageCard.css';

ImageCard.propTypes = {
  buttonText: PropTypes.string,
  cardText: PropTypes.string.isRequired,
  imageSource: PropTypes.string.isRequired,
  link: PropTypes.string,
  title: PropTypes.string.isRequired,
};

ImageCard.defaultProps = {
  buttonText: '',
  link: '',
};

function ImageCard({ buttonText, cardText, imageSource, link, title }) {
  return (
    <div className={styles.ImageCard}>
      <img className={styles.cardImage} src={imageSource} alt={title} />

      <div className={styles.cardText}>
        <h6>{title}</h6>
        <p>{cardText}</p>
        {link && (
          <Button>
            {link.startsWith('http') ? (
              <OutboundLink analyticsEventLabel={`${title} Image Card`} href={link}>
                {buttonText}
              </OutboundLink>
            ) : (
              <Link href={link}>
                {/* eslint-disable jsx-a11y/anchor-is-valid */}
                <a>{buttonText}</a>
                {/* eslint-enable jsx-a11y/anchor-is-valid */}
              </Link>
            )}
          </Button>
        )}
      </div>
    </div>
  );
}

export default ImageCard;
