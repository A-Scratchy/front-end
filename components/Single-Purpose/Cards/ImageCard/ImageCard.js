import React from 'react';
import PropTypes from 'prop-types';
import Button from 'common/components/Button/Button';
import styles from './ImageCard.css';

ImageCard.propTypes = {
  buttonText: PropTypes.string,
  cardText: PropTypes.string.isRequired,
  imageSource: PropTypes.string.isRequired,
  link: PropTypes.string,
  title: PropTypes.string.isRequired,
};

ImageCard.defaultProps = {
  buttonText: '',
  link: '',
};

function ImageCard({
  buttonText, cardText, imageSource, link, title,
}) {
  return (
    <div className={styles.ImageCard}>
      <img
        className={styles.cardImage}
        src={imageSource}
        alt={title}
      />

      <div className={styles.cardText}>
        <h6>
          {title}
        </h6>
        <p>
          {cardText}
        </p>
        {link && (
          <Button
            href={link}
            theme="primary"
          >
            {buttonText}
          </Button>
        )}
      </div>
    </div>
  );
}

export default ImageCard;
