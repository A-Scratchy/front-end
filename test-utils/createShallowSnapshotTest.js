/* eslint-env jest */
import ShallowRenderer from 'react-test-renderer/shallow';

/**
 * Used to create a snapshot test off of surface leaves of the tree.
 * Useful for regression tests on containers or larger components.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const shallowRenderer = new ShallowRenderer();
  const tree = shallowRenderer.render(Component);

  expect(tree).toMatchSnapshot();
};
