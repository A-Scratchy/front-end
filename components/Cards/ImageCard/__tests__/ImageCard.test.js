import React from 'react';
import createShallowSnapshotTest from 'test-utils/createShallowSnapshotTest';

import ImageCard from '../ImageCard';

describe('ImageCard', () => {
  it('should render with required props', () => {
    createShallowSnapshotTest(
      <ImageCard alt="Test title" imageSource="/static/images/Family1.jpg">
        <p>Testing!</p>
      </ImageCard>,
    );
  });
});
