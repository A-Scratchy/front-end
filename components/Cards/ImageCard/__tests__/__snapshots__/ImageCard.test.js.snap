// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCard it should render properly with required props 1`] = `
<div
  className="ImageCard"
>
  <img
    alt="Test title"
    className="cardImage"
    src="image.jpg"
  />
  <div
    className="cardText"
  >
    <h6>
      Test title
    </h6>
    <p />
    
  </div>
</div>
`;

exports[`ImageCard should render properly with some props assigned 1`] = `
<div
  className="ImageCard"
>
  <img
    alt="Title Tester"
    className="cardImage"
    src="image.jpg"
  />
  <div
    className="cardText"
  >
    <h6>
      Title Tester
    </h6>
    <p>
      This is the card
    </p>
    <Button
      analyticsObject={
        Object {
          "action": "Button Selected",
          "category": "Interactions",
        }
      }
      className=""
      disabled={false}
      fullWidth={false}
      onClick={[Function]}
      tabIndex={0}
      theme="primary"
      type="button"
    >
      <Link
        href="www.thisIsTheLink.com"
      >
        <a>
          Test me!
        </a>
      </Link>
    </Button>
  </div>
</div>
`;
